source/flight_software/propulsion_system.rst

```diff
@@ -0,0 +1,56 @@
+This file describes the Propulsion System driver. 
+
+PropulsionSystem class defines the interface for the propulsion system, which 
+ consists of an inner tank, Tank1, and the (outer) thrust tank, Tank2. 
+ This driver provides functionality for opening and closing valves on both tanks,
+ setting and enforcing the thrust schedule 
+ 
+ It consists of a static Tank1 (inner tank) and Tank2 (thrust tank) data objects. 
+ 
+ Specifications:
+ Tank1 valves are at numbered 0, 1
+ Tank2 valves are at numbered 0, 1, 2, 3
+ mandatory_wait_time is the duration of time after opening a valve before 
+ we can open another valve. Tank1 10*1000 ms. Tank2 is 3ms
+ 
+ State Definitions:
+ "enabled" means that the IntervalTimer for tank2 is on. This implies that:
+      - either tank2 is "scheduled (to fire)" 
+      - or tank2 has already fired
+ "scheduled (to fire)" means that tank2 currently has a scheduled start time
+ in the future
+ 
+ "start time" refers to the time in micros at which tank2 will fire
+ Prop controller is responsible for pressurizing tank1 before asking tank2 to fire
+ 
+ Dependencies: 
+ SpikeAndHold must be enabled in order to use this system
```

Comment:
Why must spikeandhold be enabled? What is spikeandhold?

Reply:
I know writing spikeandhold documentation is outside the scope of this PR, but I think it would be good if you could at least make a link to another page for Spike And Hold and leave the page blank as a TODO.

Reply:
I actually don't know why it must be enabled. This was a specification implied in the original driver code.

Reply:
For each of us, I think it's important that we take the initiative to understand the satellite as fully as we can, especially when some constraints directly interface with our own work.

In this particular case it needs to be on because the valves physically will not get the power that they need if the Spike and Hold system is not on.

Reply:
Hang on here, we can't enable spike and hold, it has no enable/disable functions, it's just a circuit. Do you mean the DCDC must be enabled? There are two DC to DC step-up converters, one for docking motor and spike and hold, and one for the reaction wheels. The docking and spike and hold one (labeled "SpikeDock" on the schematics) is the one that needs to be enabled for propulsion system to be used.